ext/TrixiConvexECOSExt.jl

```diff
@@ -0,0 +1,141 @@
+# Package extension for adding Convex-based features to Trixi.jl
+module TrixiConvexECOSExt
+
+# Required for coefficient optimization in P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using Convex
+    using ECOS
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..Convex
+    using ..ECOS
+end
```

Comment:
Instead of using everything from those packages, I recommend to follow Trixi.jl's convention to explicitly state which symbols you import. This makes it easier to understand which functionality comes from which package, e.g.,
```julia
using Convex: Variable
```
etc.